core/src/main/java/com/gentics/mesh/core/endpoint/admin/MailHandler.java

```diff
@@ -0,0 +1,126 @@
+import com.gentics.mesh.core.endpoint.handler.AbstractCrudHandler;
+import com.gentics.mesh.core.rest.MeshEvent;
+import com.gentics.mesh.core.rest.admin.mail.MailAttachmentsRequest;
+import com.gentics.mesh.core.rest.admin.mail.MailSendingRequest;
+import com.gentics.mesh.core.rest.job.JobResponse;
+import com.gentics.mesh.core.rest.job.JobStatus;
+import com.gentics.mesh.core.rest.schema.BinaryFieldSchema;
+import com.gentics.mesh.core.rest.schema.FieldSchema;
+import com.gentics.mesh.core.rest.schema.S3BinaryFieldSchema;
+import com.gentics.mesh.core.verticle.handler.HandlerUtilities;
+import com.gentics.mesh.core.verticle.handler.WriteLock;
+import com.gentics.mesh.etc.config.MailOptions;
+import com.gentics.mesh.etc.config.MeshOptions;
+import com.gentics.mesh.json.JsonUtil;
+import com.gentics.mesh.util.RxUtil;
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.mail.MailAttachment;
+import io.vertx.ext.mail.MailConfig;
+import io.vertx.ext.mail.MailMessage;
+import io.vertx.ext.mail.MailResult;
+import io.vertx.ext.mail.StartTLSOptions;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.core.file.FileSystem;
+import io.vertx.reactivex.ext.mail.MailClient;
+
+/**
+ * REST handler for job endpoint operations.
+ */
+@Singleton
+public class MailHandler extends AbstractCrudHandler<HibJob, JobResponse> {
+
+	private static final Logger log = LoggerFactory.getLogger(MailHandler.class);
+
+	private BootstrapInitializer boot;
+
+	private final PageTransformer pageTransformer;
+
+	private final Vertx vertx;
+
+	private final MeshOptions meshOptions;
+
+	@Inject
+	public MailHandler(Database db, BootstrapInitializer boot, HandlerUtilities utils, WriteLock writeLock, JobDAOActions jobActions,
+					   PageTransformer pageTransformer, Vertx vertx, MeshOptions meshOptions) {
+		super(db, utils, writeLock, jobActions);
+		this.boot = boot;
+		this.pageTransformer = pageTransformer;
+		this.vertx = vertx;
+		this.meshOptions = meshOptions;
+	}
+
+
+	/**
+	 * Invoke the job verticle for email sending.
+	 * 
+	 * @param ac
+	 */
+	public void handleInvokeMailWorker(InternalActionContext ac) {
+		utils.syncTx(ac, (tx) -> {
+			if (!ac.getUser().isAdmin()) {
+				throw error(FORBIDDEN, "error_admin_permission_required");
+			}
+			HibUser user = ac.getUser();
+			HibJob mailSendingJob = Tx.get().jobDao().enqueueMailSending(user, new Timestamp(System.currentTimeMillis()).getTime(), ac.getBodyAsString());
+
+			if(mailSendingJob.getStatus() == JobStatus.FAILED){
+				Tx.get().jobDao().enqueueMailSending(user,new Timestamp(System.currentTimeMillis()).getTime() + meshOptions.getMailOptions().getRetry(), ac.getBodyAsString());
```

Comment:
Here you have to reassign `mailSendingJob`, in order the following status check to work.